payjoin/src/ohttp.rs

```diff
@@ -237,6 +237,7 @@ impl serde::Serialize for OhttpKeys {
 
 #[derive(Debug)]
 pub enum ParseOhttpKeysError {
+    MissingOhttpKeys,
```

Comment:
I do think this is probably an error variant that belongs in url_ext and NOT ohttp.rs. Only url_ext has the concept of the parameter being "missing." This type that gets defined in url_ext will likely include whatever error variant `exp` or other fragment parameters have. Maybe it's a `pub enum FragmentError`.

Reply:
That makes sense. I had the same question while doing this, but I saw [here](https://github.com/payjoin/rust-payjoin/blob/bcbbd2b035683fd37659e6ebc2eeac2bbf8ae307/payjoin/src/uri/error.rs#L17) that `MissingPubkey` is in the same `ParseReceiverPubkeyError`. Does it make sense to move `ParseOhttpKeysError` to the `error` module?

Reply:
Hmm the url `ParseReceiverPubkeyError` that you linked to seems to be better form to me since it's specifically regarding parsing a pubkey out of the receiver pubkey from the Url fragment, and handles the variant regarding the actual pubkey contents as `InvalidPubkey(crate::hpke::HpkeError)`.

I don't think you want to *move* ParseOhttpKeysError but rather create a new variant that contains it, following the same pattern. I can see how the error in `ohttp` follows the same pattern the param error and that is confusing.

I'd make a `ParseOhttpParamError` that has an InvalidOhttpKeys(crate::ohttp::ParseOhttpKeysError) variant as well as the missing keys variant, and consider renaming ParseReceiverPubkeyError into ParseReceiverPkParamError or similar. 

I'm not 100% on the names but I think this explains the difference between the layers of abstraction and why the existing names may have raised confusion.